src/hooks/useControlledState/useControlledState.ts

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+export function useControlledState<T, C = T>(
+    value: Exclude<T, undefined>,
+    defaultValue: Exclude<T, undefined> | undefined,
+    onChange?: (v: C, ...args: any[]) => void,
+): [T, (value: C) => void];
+export function useControlledState<T, C = T>(
+    value: Exclude<T, undefined> | undefined,
+    defaultValue: Exclude<T, undefined>,
+    onChange?: (v: C, ...args: any[]) => void,
+): [T, (value: C) => void];
+export function useControlledState<T, C extends T = T>(
+    value: T,
+    defaultValue: T,
+    onUpdate?: (value: C, ...args: any[]) => void,
+) {
+    const [innerValue, setInnerValue] = React.useState(value ?? defaultValue);
+
+    const isControlledRef = React.useRef(value !== undefined);
+    const isControlled = value !== undefined;
+    React.useEffect(() => {
+        const wasControlled = isControlledRef.current;
+        if (wasControlled !== isControlled) {
+            console.warn(
+                `WARN: A component changed from ${
```

Comment:
Let's use `console.error` for better visibility. Also let's use the following format: `[useControlledState] <message>`